versioned_docs/version-2020fa/lecture10.md

```diff
@@ -3,4 +3,391 @@ id: lecture10
+### Core React Native Components
+
+Since React Native is really just React, many of the same concepts (`useState`, `props`, React Hooks, etc) still apply to React. However, instead of HTML we have `Views`. A view is the basic building block of UI in mobile development. Views can display images, hold text, handle user input, etc.
+
+Some core React Native components are:
+
+- `<View>`: A container that supports layout with flexbox, style, some touch handling, and accessibility controls. Similar to a non-scrolling `<div>`.
+- `<Text>`: Displays, styles, and nests strings of text and even handles touch events. Similar to a `<p>`
+- `<Image>`: Displays images like `<img>`
+- `<ScrollView>`: A generic scrolling container than can hold nested components and views. Similar to a `<div>`.
+- `<TextInput>`: User text input field similar to `<input type="text" />`.
+- ... and you can also define your own custom components (and use those built by the community)!
+
+### How to start a React Native Project?
+
+React Native uses `Expo` framework to develop, build, and iterate on iOS, Android and webapps. Expo provides a UI for you to view your changes and if you download the Expo app ([Android](https://play.google.com/store/apps/details?id=host.exp.exponent&hl=en_US), [iOS](https://apps.apple.com/us/app/expo-client/id982107779)) you can see those changes on your phone as well! After all, we're doing mobile development.
+
+To start a React Native project run the following:
+
+```bash
+yarn add expo-cli
+yarn expo init <project name>
+yarn start
+```
+
+`yarn expo init` is similar to `create-react-app` in that it generates boilerplate code for you.
+
+### Demo
+
+As part of the demo we built the simple TODO list app from assignment 4 in React Native! The code is here:
+
+<Tabs
+groupId="lang"
+defaultValue="ts"
+values={[
+{ label: 'TypeScript', value: 'ts', },
+{ label: 'JavaScript', value: 'js', },
+]
+}>
+
+<TabItem value="ts">
+
+```tsx title="App.tsx"
+import React, { useState } from 'react';
+import { StyleSheet, Text, View, TextInput, Button } from 'react-native';
+
+export default function App(): React.ReactElement {
+  const [item, setItem] = useState<string>('');
+  const [items, setItems] = useState<string[]>([]);
+
+  const updateItems = (): void => {
+    setItems([...items, item]);
+    setItem('');
+  };
+
+  return (
+    <View style={styles.container}>
+      <TextInput
+        placeholder="Add an item"
+        style={styles.input}
+        value={item}
+        onChangeText={(text) => setItem(text)}
+      />
+      <Button title="Add item" onPress={() => updateItems()} />
+      {items.map((i, index) => (
+        <Text key={index}> {i} </Text>
+      ))}
+    </View>
+  );
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: '#fff',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+
+  input: {
+    borderWidth: 1,
+    width: 150,
+  },
+});
+```
+
+</TabItem>
+
+<TabItem value="js">
```

Comment:
since we taught everything in TS this time, maybe we should remove the js code for future maintenance?